### DISCUSSION TITLE
How to extend the schema for custom validation?

### DISCUSSION THREAD

**Discussion by jezikk** [open-circle/valibot#215] [Q&A] [answered]
There is a basic schema:

```javascript
const schema = object({id:number()})
```

How can I **extend** the schema with some custom validation? I don't get it how to do it. :(

**Comment by fabian-hiller** [comment]
Have a look at this guide: https://valibot.dev/guides/pipelines/

**Reply by fabian-hiller** [reply]
Here is a code example:

```ts
// With validation utility
const schema1 = object({ id: number([minValue(0, 'Only positive numbers are allowed')]) });

// With custom function
const schema2 = object({ id: number([custom((input) => input >= 0, 'Only positive numbers are allowed')]) });
```

**Reply by jezikk** [reply]
> Have a look at this guide: https://valibot.dev/guides/pipelines/

There is no info how to extend the existing schema with only custom validation. I can't figure out how to access the pipeline of existing schema instance.

**Reply by jezikk** [reply]
> Here is a code example:
> 
> 
> 
> ```ts
> 
> // With validation utility
> 
> const schema1 = object({ id: number([minValue(0, 'Only positive numbers are allowed')]) });
> 
> 
> 
> // With custom function
> 
> const schema2 = object({ id: number([custom((input) => input >= 0, 'Only positive numbers are allowed')]) });
> 
> ```

My question is how I can add a new pipeline validation into schema2 instance?

**Reply by fabian-hiller** [reply, marked as answer]
> There is no info how to extend the existing schema with only custom validation. I can't figure out how to access the pipeline of existing schema instance.

We work on this feature in PR #211. At the moment it is not possible to access the pipeline of a schema afterwards. As a workaround, you can store the pipeline in a separate variable to reuse it in other places.

**Reply by jezikk** [reply]
> > There is no info how to extend the existing schema with only custom validation. I can't figure out how to access the pipeline of existing schema instance.
> 
> 
> 
> We work on this feature in PR #211. At the moment it is not possible to access the pipeline of a schema afterwards. As a workaround, you can store the pipeline in a separate variable to reuse it in other places.

OK, thanks.
